docs/migration-guide-api-with-lambda.md

```diff
@@ -0,0 +1,142 @@
+# Migration Guide (GuApiLambda & GuApiGatewayWithLambdaByPath)
+
+This document assumes you have completed initial setup and other steps described
+in the [Migration Guide](./migration-guide.md).
+
+---
+
+For apps where avoiding downtime is critical, we recommend a _dual-stack
+migration strategy_. The idea is to duplicate the key parts of the stack using
+the pattern, test, and, once confident, switch DNS to point to the new stack.
+
+**Stage 1: dual-stack**
+
+```mermaid
+graph TD
+    A(example.gutools.co.uk)
+    A --> B[Legacy API Gateway instance & Lambdas]
+    A .-> C[New API Gateway instance & Lambdas]
+```
+
+**Stage 2: switch DNS**
+
+```mermaid
+graph TD
+    A(example.gutools.co.uk)
+    A .-> B[Legacy API Gateway instance & Lambdas]
+    A --> C[New API Gateway instance & Lambdas]
+```
+
+**Stage 3: cleanup**
+
+```mermaid
+graph TD
+    A(example.gutools.co.uk)
+    A --> C[New API Gateway instance & Lambdas]
+```
+
+## Stage 1: the new stack
+
+_This stage creates a new version of your app's infrastructure, according to
+CDK’s best practices. Note that some resources (e.g. Dynamo tables) will not be
+duplicated as part of the pattern._
+
+_At the end of this stage, you should have a functional copy of your
+application, which can be used for testing purposes. Riff-Raff will have been
+configured to deploy to both versions of your infrastructure, so you can operate
+in this state for as long as you need to._
+
+1. Identify the correct GuCDK pattern
+
+    If your app serves all requests via a single Lambda function, you will need [`GuApiLambda`](https://guardian.github.io/cdk/classes/patterns.GuApiLambda.html).
+
+    If your app routes different requests to different Lambda functions, you will need [`GuApiGatewayWithLambdaByPath`](https://guardian.github.io/cdk/classes/patterns.GuApiGatewayWithLambdaByPath.html).
+
+2. Create an instance of the pattern
+
+    Use your existing stack as a guide. For example, you will likely need to
+    create some custom IAM permissions and add them to your
+    [Lambda](https://guardian.github.io/cdk/classes/constructs_lambda.GuLambdaFunction.html).
+
+    There's an example of this [here](https://github.com/guardian/tag-janitor/blob/9e2222d7cea6b37a48e5327efecf7b543b81af15/cdk/lib/cdk-stack.ts#L57-L62).
+
+3. Update your `riff-raff.yaml` to deploy application code to your new Lambda function(s)
+
+    As we will be using a dual-stack migration strategy, we want Riff-Raff to update the application code for old (CFN-defined)
+    and new (GuCDK-defined) versions of the Lambda function(s) as part of every deployment.
+
+    There's an example of this [here](https://github.com/guardian/mobile-save-for-later/pull/65/files#diff-f1f55b65ced3f9ca1a6a068f04d6dc6241689cef94fce7c0cb62422495ff4534).
+
+    Note that you **must** use the `functionNames` parameter for this to work.
+    The alternative strategy (which uses `lookupByTags`), will not work with the dual-stack approach.
```

Comment:
Yes, I think this would be possible.

However using the names actually seems a lot more straightforward than following the slightly awkward process that we use to cope with having two ASGs with the same tags. See [here](https://github.com/guardian/cdk/blame/main/docs/migration-guide-ec2.md#L86-L94) & [here](https://github.com/guardian/cdk/blame/main/docs/migration-guide-ec2.md#L187-L189). 

So I'm unsure if we really want to introduce another way of doing things here ([especially as `lookupByTags` seems to be false by default](https://riffraff.gutools.co.uk/docs/magenta-lib/types#awslambda) anyway).